### ISSUE TITLE
Fix S3433: Rule should not report on non-public test methods for xUnit

### ISSUE THREAD

**Issue by Evangelink** [SonarSource/sonar-csharp#1262]
### Description

`XUnit` does handle non-`public` methods for tests.

### Repro steps

```
[Fact]
private void TestMethod1()
{
	Assert.True(false);
}
```
